src/dodona/junit/MultiMessageWriter.java

```diff
@@ -0,0 +1,118 @@
+package dodona.junit;
+
+import dodona.feedback.Format;
+import dodona.feedback.Message;
+import dodona.feedback.Permission;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Predicate;
+
+/**
+ * Feedback message writer that allows multiple messages and rewinding of
+ * previously sent messages.
+ */
+public class MultiMessageWriter implements TestRule {
```

Comment:
Doesn't this class replace all MessageWriter functionality? Then maybe we should actually replace it (if that's backwards compatible).

Reply:
It does not, since this class is not a PrintStream. You can only append Messages as a whole (as opposed to the MessageWriter). It is however possible to merge both functionalities in a backwards compatible way, but this would probably be ugly as it would require the caller of this class to use methods like `finishMessage(); newMessage();`. It is however a possibility if desired.